File: java/core/src/main/java/com/github/fernthedev/lightchat/core/packets/handshake/ConnectedPacket.java
package com.github.fernthedev.lightchat.core.packets.handshake;

import com.github.fernthedev.lightchat.core.VersionData;
import com.github.fernthedev.lightchat.core.packets.Packet;
import com.github.fernthedev.lightchat.core.packets.PacketInfo;
import lombok.AccessLevel;
import lombok.Getter;
import lombok.NonNull;

/**
 * Final packet sent in the handshake
 */
@Getter
@PacketInfo(name = "CONNECTED_PACKET")
public class ConnectedPacket extends Packet {
    @NonNull
    private String name;

    @Getter(AccessLevel.NONE)
    @NonNull
    private String os;

    @NonNull
    private VersionDataString versionData;

    public VersionData getVersionData() {
        return new VersionData(versionData.getVersion(), versionData.getMinVersion());
    }



//    @NonNull
//    private UUID uuid;

    public ConnectedPacket(@NonNull String name, @NonNull String os, VersionData versionData) { //, @NonNull UUID uuid) {
        this.name = name;
        this.os = os;
        this.versionData = new VersionDataString(versionData.getVersion().toString(), versionData.getMinVersion().toString());
//        this.uuid = uuid;
    }


    @Override
    public String toString() {
        return "ConnectedPacket{" +
                "name='" + name + '\'' +
                ", os='" + os + '\'' +
                '}';
    }


    public String getOS() {
        return os;
    }
}

File: java/core/src/main/java/com/github/fernthedev/lightchat/core/packets/handshake/ConnectedPacket.kt
package com.github.fernthedev.lightchat.core.packets.handshake

import com.github.fernthedev.lightchat.core.VersionData
import com.github.fernthedev.lightchat.core.packets.PacketInfo
import com.github.fernthedev.lightchat.core.packets.PacketJSON

/**
 * The final packet sent in the handshake
 */
@PacketInfo(name = "CONNECTED_PACKET")
class ConnectedPacket private constructor(
    val name: String, val os: String, val versionData: VersionDataString,
    val langFramework: String
) : PacketJSON() {

    fun getVersionData(): VersionData {
        return VersionData(versionData.version, versionData.minVersion)
    }

    constructor(name: String, os: String, versionData: VersionData, langFramework: String) : this(
        name,
        os,
        VersionDataString(versionData.version.toString(), versionData.minVersion.toString()),
        langFramework
    )

    override fun toString(): String {
        return "ConnectedPacket{" +
                "name='" + name + '\'' +
                ", os='" + os + '\'' +
                '}'
    }
}